src/lib/geoip/legal-name-normalization.ts

```diff
@@ -0,0 +1,175 @@
+import fs from 'node:fs';
+import ascii from 'any-ascii';
+import csvParser from 'csv-parser';
+import is from '@sindresorhus/is';
+import _ from 'lodash';
+
+// See https://github.com/jsdelivr/globalping/issues/383
+// The CSV file is from https://www.gleif.org/en/lei-data/code-lists/iso-20275-entity-legal-forms-code-list
+const FILENAME = '2023-09-28-elf-code-list-v1.5.csv';
+
+const ADDITIONAL_SUFFIXES = [
+	{ name: 'Limitada', abbr: [ 'LDA' ] }, // Portugal
+];
+
+let namesPattern: RegExp;
+let abbrsPattern: RegExp;
+
+type CsvLegalFormRow = {
+	elfCode: string;
+	countryOfFormation: string;
+	countryCode: string;
+	jurisdictionOfFormation: string;
+	countrySubDivisionCode: string;
+	entityLegalFormNameLocal: string;
+	language: string;
+	languageCode: string;
+	entityLegalFormNameTransliterated: string;
+	abbreviationsLocal: string;
+	abbreviationsTransliterated: string;
+	dateCreated: string;
+	elfStatus: string;
+	modification: string;
+	modificationDate: string;
+	reason: string;
+};
+
+export const normalizeLegalName = (name: string) => {
+	if (!namesPattern || !abbrsPattern) {
+		throw new Error('Legal name normalization is not initialized.');
+	}
+
+	return name.trim()
+		// Normalize "trading as" names, e.g., "Matteo Martelloni trading as DELUXHOST" => "DELUXHOST"
+		.split(/\s+trading as\s+/i).at(-1)!
+		// Apply the main cleanup patterns.
+		.replace(namesPattern, '').trim()
+		.replace(abbrsPattern, '').trim()
```

Comment:
Looks like regex search might take a lot of time: https://github.com/jsdelivr/globalping/actions/runs/17804117023/job/50611441251#step:7:45 
It takes even more localy.

Maybe we can add more variants to `legalFormsLong` and `legalFormsAbbr`? E.g. `llc`, `l.l.c.` and `l l c`. Then split the network value by spaces, commas, parens, etc. and use `has()` for every part starting from the end?

Reply:
> Then split the network value by spaces, commas, parens, etc. and use has() for every part starting from the end?

This would be much slower with the high number of variants. The regexes are generaly the fastest way to check for many possible values in V8.